docs/secret-template.md

```diff
@@ -1,56 +1,124 @@
-### Secret Template
+## SecretTemplate
 
-By default generated secrets have predefined set of keys. In a lot of cases `Secret` consumers expect specific set of keys within `data`. To control customization of keys within a secret, you can configure use secret template.
+As of v0.9.0+, the `SecretTemplate` is now avaliable under the `secretgen.carvel.dev` API group.
 
-Fields:
+Secrets are a common method of encapsulating and inputing sensitive data into an API via reference, or to a process via volume mounting.  In order to make compatible Secrets for these use cases, it often requires imperatively reading of other resources (including other secrets) and imperatively creating the compatible secret using these resources' information.  `SecretTemplate` aims to provide a declarative way of doing this process.
```

Comment:
I think this focuses a little too much on the usecase, perhaps we could reorder slightly, open with what and then expand on the why after? 

Something like:

```suggestion
The `SecretTemplate` API allows new Secrets to be composed from data residing in existing Kubernetes Resources, including other Secrets.

Secrets are a common method of encapsulating and inputing sensitive data into other Kubernetes Resource via a reference, or to a process via [volume mounting](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod). However the required information may be contained in a number of other Kubernetes Resources. 
```

Reply:
Also seems reasonable!